### ISSUE TITLE
Wrong method using headers on httpGet HealthCheck

### ISSUE THREAD

**Issue by i2dcarrasco** [kubernetes/kubernetes#94682]
Hello,

**What happened**:
When I use the httpGet health check combined with a header, my pods gives an HTTP 400 error. When I remove the headers it works (I receive a 403 error, but is correct because does not have the host header).

**What you expected to happen**:
A normal GET petition even with headers.

**How to reproduce it (as minimally and precisely as possible)**:
Any pod configuration with a simple HTTP readinessProbe or livenessProbe, including headers.
```
httpGet:
  path: "/pub/health_check.php"
  port: 8080
  httpHeader:
    name: "HOST"
    value: "www.hostname.com"
```

**Anything else we need to know?**:
I've got this error on a pod with Nginx, and also on another with Varnish. The varnish logs shows as method "GET%00" when I use headers, and I think that's why is marked as incorrect method.

**Environment**:
- Kubernetes version (use `kubectl version`): v1.15.12-gke.2
- Cloud provider or hardware configuration: GKE
- OS (e.g: `cat /etc/os-release`): Google Container Optimized OS (image gke-11512-gke2-cos-77-12371-227-0-v200601-pre)
- Kernel (e.g. `uname -a`): I'm not sure, I've no access to the instance.
- Others: I'm using terraform to configure the pod, but I remember to see the same problem using kubectl.

Best regards.

**Comment by zegl** [comment]
The header field is called `httpHeaders`, not `httpHeader`, and is a list of maps, not a map. [[docs]](https://v1-15.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.15/#httpgetaction-v1-core)

The correct syntax should be:

```
httpGet:
  path: "/pub/health_check.php"
  port: 8080
  httpHeaders:
  - name: "HOST"
    value: "www.hostname.com"
```

Is the issue reproducible after you've made these changes?

**Comment by i2dcarrasco** [comment]
Hello, thanks for you answer.

Sorry for that, I've copied the Terraform config that is a bit different. I've tried to correct it to make it like the resulting yaml, but I had no time to check the correct syntax (I changed the check to TCP and I had no access to the output). Anyway, I've checked the syntax several times (terraform alert me if syntax is incorrect), and also the resulting yaml with the describe command, and was right. Also the problem seems to be the Headers because the same syntax without headers works perfecly.

Anyway, tomorrow I'll try again to give a try, but I'm sure that was correct.

Best regards.
